### PR TITLE
watchOS support for FirebaseAuth.

### PR DESCRIPTION
**Pull request by ryanwilson**
This is a redo of #5585, which was a re-implementation of #4632. This
ignores any changes to GoogleUtilities, those will be done in a later
PR to create a common KeyedArchiver utility.

Credit goes to @hohteri for doing this work in the first place! Let's finally land this thing 😄

### CODE REVIEW

**Comment by hohteri** [comment]
Boom

**Comment by ryanwilson** [comment]
@hohteri apologies for the long, winding road that this path has taken :P but once tests pass and everything right hopefully we can land this today. Mind adding a comment again saying `@googlebot I consent` for the bots to be happy?

**Comment by hohteri** [comment]
@googlebot I consent

File: .github/workflows/auth.yml [hunk 1]
```diff
@@ -104,7 +104,8 @@ jobs:
     runs-on: macos-latest
     strategy:
       matrix:
-        target: [ios, tvos --skip-tests, macos --skip-tests] # The macos and tvos tests can hang
+        # The macos and tvos tests can hang, and watchOS doesn't have tests.
+        target: [ios, tvos --skip-tests, macos --skip-tests, watchos --skip-tests]
```
Comment (paulb777) [inline .github/workflows/auth.yml:108]: Please temporarily disable schedule or run manually.

Thanks for adding!

**Pull request merged**